### DISCUSSION TITLE
Why does client::Connection not require quic::Send/RecvDatagramExt

### DISCUSSION THREAD

**Discussion by gopakumarce** [hyperium/h3#226] [General]
ClientConnection only requires quic::Connection, any reason to exclude the send/recv datagram from it ? It would be nice to be able to use the client::Connection to also send and receive datagrams and hence the question

**Comment by gopakumarce** [comment]
I see that 22da9387f19d724852b3bf1dfd7e66f0fd45cb81 added send_datagran/recv_datagram on the server side, I guess it just needs to be added on the client side and there is no other specific reason ? If so ill float a PR for that

**Comment by Ruben2424** [comment]
The `send_datagran`/`recv_datagram` functions where added in order to support WebTransport.
But they are not part of the http3 spec. 
Instead the functionality to send datagrams is part of a different [spec](https://datatracker.ietf.org/doc/rfc9297/ ).
In my opinion this should be in a different crate like WebTransport or at least behind a feature gate.

**Reply by gopakumarce** [reply]
I see,  @Ruben2424  so what you are saying is that even the send_datagram/recv_datagram in h3/src/server.rs does not really belong there and should be moved elsewhere ?

**Reply by Ruben2424** [reply]
If you ask me, yes, it belongs in a separate crate. 
I already started to move the datagram pieces to a separate crate #199 but hadn't time to finish the work.
If you want, feel free to continue the work and add client api.

**Reply by gopakumarce** [reply]
Ok thx @Ruben2424 .. let me take a look at your changes and I'll continue the work
